src/reader.c

```diff
@@ -209,17 +209,16 @@ static int Reader_init(hiredis_ReaderObject *self, PyObject *args, PyObject *kwd
             return -1;
 
     self->encoding = encoding;
-    if(errors) {
-        if (strcmp(errors, "strict") != 0 &&
-            strcmp(errors, "replace") != 0 &&
-            strcmp(errors, "ignore") != 0 &&
-            strcmp(errors, "backslashreplace") != 0) {
-
-            PyErr_SetString(PyExc_LookupError,
-                            "if specified, errors must be one of "
-                            "{'strict', 'replace', 'ignore', 'backslashreplace'}");
+    if (errors) {   // validate that the error handler exists, raises LookupError if not
+        PyObject *codecs, *result;
```

Comment:
Please can you explain what's the problem?

If it's the C++ style comment, I note that there are several other instances of this in the same file.  Otherwise I can't see a compatibility problem.

There are no errors or warnings when reader.c is compiled with `-std=c99 -Wpedantic` or `-std=c11 -Wpedantic`

If I compile reader.c with `-std=c89 -Wpedantic`, it does not object to any of the code I added.  But it *does* trip up on a number of unrelated instances of non-compliance with the older ISO C89/C90/ANSI standard: e.g.

```
warning: ISO C90 does not support ‘long long’
error: C++ style comments are not allowed in ISO C90
```

Reply:
All declarations need to be put at the function beginning.

Reply:
I tested declarations at the top of if block and it turned out that i was wrong.
And from those // comments we're not stick to C89 either.

Reply:
Ah I see.  Declarations at the top of a compound expression are permitted all the way back to K&R 2nd edition (draft-proposed ANSI C) - I still have this on my shelf.  It's declaring variables in the *middle* of a block which is a more recent extension inherited from C++:

```
{
    char *a;
    foo();
    char *b;   /* << this */
    ...
}
```